### DISCUSSION TITLE
REST API

### DISCUSSION THREAD

**Discussion by jcarlosfernandez** [papermerge/papermerge-core#15] [General]
Hello, I am programming a client for papermerge and I would like to know the status of the programming of the rest api. From what I see for now, only the part of sending documents is done, is that correct? 

Many thanks for your work.

**Comment by ciur** [comment]
Latest REST API documentation is available [here](https://docs.papermerge.io/REST%20API/index.html). I've documented so far following endpoints: users, documents, folders, nodes, preferences, tokens, search, auth.

In [docker/readme.md](https://github.com/papermerge/papermerge-core/blob/master/docker/README.md) you can find ready to use docker compose file to quickly start all required services locally. You are probably interested only in [docker/prod-backend-only.yml](https://github.com/papermerge/papermerge-core/blob/master/docker/README.md#prod-backend-onlyyml), which if you start, you'll can point your browser to http://localhost:8000/api/. Click login in upper right corner, provide same credentials as ``DJANGO_SUPERUSER_USERNAME`` and ``DJANGO_SUPERUSER_PASSWORD`` from environment file and you will be able to play with browsable REST API.

I will update soon docker section of the documentation which will explain how to setup and use latest development version.

**Comment by jcarlosfernandez** [comment]
Thanks a lot, I'll try to put It un production.

**Reply by ciur** [reply]
I've just updated online documentation with [how to setup backend only with docker compose](https://docs.papermerge.io/Installation/docker.html).

**Comment by jcarlosfernandez** [comment]
Awesome, thanks!
